src/theory/quantifiers/sygus_process_conj.h

```diff
@@ -15,26 +15,234 @@
+
+ private:
+  /** the synth fun associated with this */
+  Node d_synth_fun;
+  /** properties of each argument */
+  std::vector<CegConjectureProcessArg> d_arg_props;
+  /** variables for each argument type of f
+   *
+   * These are used to express templates for argument
+   * invariance, in the data member
+   * CegConjectureProcessArg::d_template.
+   */
+  std::vector<Node> d_arg_vars;
+  /** map from d_arg_vars to the argument #
+   * they represent.
+   */
+  std::unordered_map<Node, unsigned, NodeHashFunction> d_arg_var_num;
+  /** check match
+   * This function returns true iff we can infer:
+   *   cn * { x -> n_arg_map[d_arg_var_num[x]] | x in d_arg_vars } = n
+   * In other words, cn and n are equivalent
+   * via the substitution mapping argument variables to terms
+   * specified by n_arg_map. The rewriter is used for inferring
+   * this equivalence.
+   *
+   * For example, if n_arg_map contains { 1 -> t, 2 -> s }, then
+   *   checkMatch( x1+x2, t+s, n_arg_map ) returns true,
+   *   checkMatch( x1+1, t+1, n_arg_map ) returns true,
+   *   checkMatch( 0, 0, n_arg_map ) returns true,
+   *   checkMatch( x1+1, s, n_arg_map ) returns false.
+   */
+  bool checkMatch(Node cn,
+                  Node n,
+                  std::unordered_map<unsigned, Node>& n_arg_map);
+  /** infer definition
+   *
+   * In the following, we say a term is a "emplate
+   * definition" if its free variables
+   * are a subset of d_arg_vars.
+   *
+   * If this function returns a non-null node ret, then
+   *   checkMatch( ret, n, term_to_arg_carry^-1 ) returns true.
+   * and ret is a template definition.
+   *
+   * The free variables for all subterms of n are stored in
+   * free_vars. The map term_to_arg_carry is injective.
+   *
+   * For example, if term_to_arg_carry contains { t -> 1, s -> 2 } and
+   * free_vars is { t -> {y}, r -> {y}, s -> {}, q -> {}, ... -> {} }, then
+   *   inferDefinition( 0, term_to_arg_carry, free_vars )
+   *     returns 0
+   *   inferDefinition( t, term_to_arg_carry, free_vars )
+   *     returns x1
+   *   inferDefinition( t+s+q, term_to_arg_carry, free_vars )
+   *     returns x1+x2+q
+   *   inferDefinition( t+r, term_to_arg_carry, free_vars )
+   *     returns null
+   * 
+   * Notice that multiple definitions are possible, e.g. above:
+   *  inferDefinition( s, term_to_arg_carry, free_vars )
+   *    may return either s or x2     
+   * TODO (#1210) : try multiple definitions?
+   */
+  Node inferDefinition(
+      Node n,
+      std::unordered_map<Node, unsigned, NodeHashFunction>& term_to_arg_carry,
+      std::unordered_map<Node,
+                         std::unordered_set<Node, NodeHashFunction>,
+                         NodeHashFunction>& free_vars);
+  /** Assign relevant definition
+   *
+   * If def is non-null,
+   * this function assigns def as a template definition
+   * for the (currently irrelevant) arguments that
+   * occur in a term of the form f(
```

Comment:
There seems to be something missing: ... of the form f( ?

Reply:
Yes, I'll fix.